pkg/catalogv2/helm/client.go

```diff
@@ -0,0 +1,37 @@
+package helm
+
+import (
+	"github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+type Client struct {
+	actRun           actionRunFunc
+	newList          newList
+	restClientGetter genericclioptions.RESTClientGetter
+}
+
+func NewClient(restClientGetter genericclioptions.RESTClientGetter) *Client {
+	return &Client{restClientGetter: restClientGetter, actRun: runAction, newList: action.NewList}
+}
+
+func (c *Client) ListReleases(namespace, name string, stateMask action.ListStates) ([]*release.Release, error) {
+	helmCfg := &action.Configuration{}
+	if err := helmCfg.Init(c.restClientGetter, namespace, "", logrus.Infof); err != nil {
+		return nil, err
+	}
+	l := c.newList(helmCfg)
+	l.Filter = "^" + name + "$"
+	l.StateMask = stateMask
+	return c.actRun(l)
+}
+
+type actionRunFunc func(*action.List) ([]*release.Release, error)
+
+type newList func(*action.Configuration) *action.List
```

Comment:
nit: it's fine if you leave this, but you could just put these signatures in the Client struct, not necessary to make them their own types.